### PR TITLE
If T tag in dss file does not exist, set issue date to dPart of the d…

### PR DESCRIPTION
**Pull request by dvdhoz**
…ss pathname. Set the combo box enable to false when T tag is missing.

### CODE REVIEW

File: ensemble-view/src/main/java/hec/ensembleview/DatabaseHandlerService.java [hunk 1]
```diff
@@ -63,6 +62,10 @@ public void setDbHandlerZonedDateTime(ZonedDateTime zdt) {
         }
     }
 
+    public boolean getIsIssueDateAvailableForDss() {
```
Comment (ktarbet) [inline ensemble-view/src/main/java/hec/ensembleview/DatabaseHandlerService.java:65]: Having 'DSS' in the method name raises some questions:

* Is the DatabaseHandlerService intended as a interface? 
* What is different besides the class EnsembleDatabase interface

**Review by ktarbet — COMMENTED** [review]
It looks like the fallback to 'use-D-part' code is now in both Database and Catalog.

Consider having users opt-in/ask for a change in behavior instead of changing on the fly.  Here that could mean a property or method such as  IssueDateCallback( dsspath-> {code} )

Code that changes, doing special things based on the data-storage-internals, is a slippery slope that looks flexible at first, then can become confusing to future developers, and have subtle bugs as you add more 'flexibility' for different DSS files.

**Comment by dvdhoz** [comment]
Removed the estimate issued date logic from DssDatabase so it only happens in Catalog.  For this first attempt, I am struggling to figure out how to best implement the opt in behavior.

**Review by joshuarwillis — CHANGES_REQUESTED** [review]
I tested the latest version of the branch using the final project files from the linked tutorial and guide below. Since the T tag of the f-part doesn't exist, the current time is used. The original plan was to use the d-part in this scenario, right?

Currently, if the DSS file is missing the T tag and the current time is used instead, the plot does not update correctly whenever the "Location" combo-box selection is changed. In the below figure, flow is selected but the plot still shows moisture deficit. I noticed that plots do update correctly if the chosen file does contain a T tag.
image



https://www.hec.usace.army.mil/confluence/hmsdocs/hmsguides/ensemble-analysis-simulations-in-hec-hms/creating-an-ensemble-analysis-compute

**Review by joshuarwillis — APPROVED** [review]

**Review by joshuarwillis — APPROVED** [review]

**Pull request merged**